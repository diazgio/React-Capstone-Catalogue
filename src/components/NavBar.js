/* eslint-disable arrow-body-style */
import React from 'react';
import { Link } from 'react-router-dom';

const NavBar = () => {
  return (
    <div className="navbar">
      <h1>Pokedex 2021</h1>
      <ul>
        <li>
          <Link className="link" to="/">Home</Link>
        </li>
      </ul>
    </div>
  );
};

export default NavBar;
