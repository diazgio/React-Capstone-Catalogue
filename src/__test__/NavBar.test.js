import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import NavBar from '../components/NavBar';

describe('Creates a component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Router>
        <NavBar />
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders a component', () => {
    render(<Router><NavBar /></Router>);
    const element = screen.getByText('Pokedex First Gen');
    expect(element).toBeInTheDocument();
  });
});
