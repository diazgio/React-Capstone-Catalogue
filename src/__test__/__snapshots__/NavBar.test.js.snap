// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates a component renders correctly 1`] = `
<div
  className="navbar"
>
  <h1>
    Pokedex First Gen
  </h1>
  <ul>
    <li>
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="link"
        href="/types"
        onClick={[Function]}
      >
        Types
      </a>
    </li>
  </ul>
</div>
`;
